### DISCUSSION TITLE
How integrate a custom IDE / editor?

### DISCUSSION THREAD

**Discussion by OCram85** [loft-sh/devpod#1408] [Q&A] [answered]
Hey folks, I am very glad that i came across this project!

I'm currently working on a custom VSCode distribution: https://arkanum.dev.
Is there any way I can start this container as workspace and forward the ports to access via Browser?

I tested it with the pre build image option but the forwarding doesn't seem to work.

#### 📝 Additonal docs

- Arkanum PR for devcontainer support: https://gitea.ocram85.com/arkanum/arkanum/pulls/128
- pre build image sources:
  - `ocram85/arkanum:pull_128`
  - `ghcr.io/ocram85/arkanum:pull_128`
  - `gitea.ocram85.com/arkanum/arkanum:pull_128`
  - `codeberg.org/arkanum/arkanum:pull_128`

**Comment by abdulari** [comment, marked as answer]
hey.
IIRC, DevPod works by running ssh server, and exposing it for ssh tunneling.

so, if your editor can access remote machine via SSH, devpod container it should able to connect with your editor.
try `ssh your-workspace-name.devpod` in your editor or terminal.

![image](https://github.com/user-attachments/assets/8de527e4-485b-44aa-b0fb-ff9310717151)
